src/main/docs/guide/views/templates/jstachio.adoc

```diff
@@ -0,0 +1,43 @@
+Micronaut Views JStachio includes api:views.jstachio.JStachioViewsRenderer[] which uses
+https://github.com/jstachio/jstachio[JStachio] A type-safe compiled Mustache based template engine.
+
+Add the `micronaut-views-jstachio` dependency to your classpath.
+
+dependency:micronaut-views-jstachio[groupId="io.micronaut.views"]
+
+The example shown in the <<views, Views>> section, could be rendered with the following JStachio 
+annotated model and template:
+
+[source,java]
+.src/main/java/myapp/HomeModel.java
+----
+//Normally you would put the JStacheConfig annotation on package-info or module-info
+//but it can be placed on the model class directly
+@JStacheConfig(using = MicronautJStacheConfig.class) 
+@JStache(path = "home")
+public record HomeModel(@Nullable String username, boolean loggedIn) {
+
+}
+----
+
+[source,html]
+.src/main/resources/views/home.mustache
+----
+include::{testsviewsJstachio}/resources/views/home.mustache[]
```

Comment:
JStachio is a mustache engine. I am not an ascii doc pro and wasn't sure if mustache syntax is supported. Should I change it to [source,mustache] ?

Reply:
```java
    @View(JSTACHIO_VIEW)
    @Get("/pojo")
    public HttpResponse<Person> pojo() {
        return HttpResponse.ok(new Person("sdelamo", true));
    }
```

See JStachio is different than other templating engines in that the model is coupled with the template and you cannot just use `Map<String,Object>`. Because of this the string literal view name is not important to JStachio however it is for Micronaut to resolve which ViewRenderer to pick. So I picked the view name "jstachio".

Consequently if someone wanted to use JMustache, or Mustache.java with JStachio they should be fine as in those engines they view name would actually be the template.

Ideally the `ViewRenderer` would have a contract of:

```java
boolean exists(String viewName, T data){
// here jstachio would ignore the viewname and check the data if it is a jstache model.
}
```

Perhaps there is some `Encoder` like interface I should use instead (which is how this is implemented in Spring)?

That is instead of the returning JSON I would encode to HTML and thus the whole `@View` can be omitted but I'm not sure how to do that in Micronaut yet.

Reply:
It looks like `MessageBodyWriter` is the one I want and then I annotate with `@Produces` (html in this case) I think.

In theory if I register the above one can then do:

    // @View is no longer needed
    @Get("/pojo")
    public HttpResponse<Person> pojo() {
        return HttpResponse.ok(new Person("sdelamo", true));
    }

Person then could never be rendered as JSON (if its annotated with `@JStache`) but my experience with this is that template models require different fields than REST API models and composition can always be done e.g. `new PersonJStache(new Person(...))`.

@sdelamo should I include a `MessageBodyWriter` implementation here or is that not a good idea?